web/account_controller.go

```diff
@@ -0,0 +1,79 @@
+package web
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/asdine/storm"
+	"github.com/asdine/storm/q"
+	"github.com/gin-gonic/contrib/sessions"
+	"github.com/gin-gonic/gin"
+	"github.com/manyminds/api2go/jsonapi"
+	"github.com/smartcontractkit/chainlink/services"
+	"github.com/smartcontractkit/chainlink/store/models"
+	"github.com/smartcontractkit/chainlink/store/presenters"
+	"github.com/smartcontractkit/chainlink/utils"
+)
+
+// AccountController manages the current Session's User Account.
+type AccountController struct {
+	App *services.ChainlinkApplication
+}
+
+func (c *AccountController) getCurrentSessionID(ctx *gin.Context) (string, error) {
+	session := sessions.Default(ctx)
+	sessionID, ok := session.Get(SessionIDKey).(string)
+	if !ok {
+		return "", errors.New("unable to get current session ID")
+	}
+	return sessionID, nil
+}
+
+func (c *AccountController) clearNonCurrentSessions(sessionID string) error {
+	var sessions []models.Session
+	err := c.App.Store.Select(q.Not(q.Eq("ID", sessionID))).Find(&sessions)
+	if err != nil && err != storm.ErrNotFound {
+		return err
+	}
+
+	for _, s := range sessions {
+		err := c.App.Store.DeleteStruct(&s)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (c *AccountController) updateUserPassword(user *models.User, newPassword string) error {
+	hashedPassword, err := utils.HashPassword(newPassword)
+	if err != nil {
+		return err
+	}
+	user.HashedPassword = hashedPassword
+	return c.App.Store.Save(user)
+}
+
+// UpdatePassword changes the password for the current Account.
+func (c *AccountController) UpdatePassword(ctx *gin.Context) {
```

Comment:
Seems non-RESTful. Could we just make this `Update` for now, since there's nothing else to update on an account?

Reply:
I guess that's a matter of opinion, I'm following the strategy of `singleton name in path` = singleton resource, i.e. `/account` is the current account, `/password` is the password for the account, this has slightly different behavior to a regular update endpoint i.e. it needs the old password, so putting in `/account/update` will almost certainly require breaking it off later.